### ISSUE TITLE
Issue downloading an invoice

### ISSUE THREAD

**Issue by ajouve** [XeroAPI/xero-node#261]
Hi 

From time to time we have a corrupted document when we download an invoice

Here the code extract in nodejs

```js
const downloadInvoice = invoiceId => {
    const savePath = `${__dirname}/../tmp/${invoiceId}.pdf`;
    return xero.invoices.savePDF({InvoiceID: invoiceId, savePath})
    .then(() => { 
        const data = fs.readFileSync(savePath);
        fs.unlinkSync(savePath);
        return data;
    })
}
```

We have the issue on the first download, we will add an hot fix calling twice xero.invoices.savePDF but this is not a long term solution.
The second download is not always working

Attached a corrupted invoice 

Kind regards

[FA6404.pdf](https://github.com/XeroAPI/xero-node/files/3177595/FA6404.pdf)

**Comment by ajouve** [comment]
Little update,

This code works

```js
    const downloadInvoice = invoiceId => {
        const savePath = `${__dirname}/../tmp/${invoiceId}.pdf`;
        return xero.invoices.savePDF({InvoiceID: invoiceId, savePath})
        .then(() => sleep(1000))
        .then(() => {
            const data = fs.readFileSync(savePath);
            fs.unlinkSync(savePath);
            return data;
        })
    }

    const sleep = time => new Promise(resolve => {
        setTimeout(() => {
            resolve()
        }, time);
    })
```

Seems that when the then is called, the file is not fully downloaded

**Comment by stansv** [comment]
So the real issue here is that `savePDF` resolves while operation not fully completed, right?

**Comment by ajouve** [comment]
@stansv yes s stream or a buffer will be much more flexible.

**Comment by ajouve** [comment]
Why did you close it ?

**Comment by iamam34** [comment]
@ajouve I closed #250 because it is a duplicate of #261. This issue #261 is still open.

I agree that this would be a great feature. I don't have time to look at it right now; feel free to open a PR by modifying the `generateAttachmentsEndpoint()` helper method at https://github.com/XeroAPI/xero-node/blob/00fc5e3f896156b4af55e83660e600a09422fa74/src/AccountingAPIClient.ts#L41.

**Comment by SerKnight** [comment]
Hi all - just linking that this functionality is active in the most recent version of the xero-node repo: https://github.com/XeroAPI/xero-node-oauth2-app/blob/master/src/app.ts#L1141
